pkg/cache/internal/deleg_map.go

```diff
@@ -57,25 +57,25 @@ func NewInformersMap(config *rest.Config,
 }
 
 // Start calls Run on each of the informers and sets started to true.  Blocks on the stop channel.
-func (m *InformersMap) Start(stop <-chan struct{}) error {
-	go m.structured.Start(stop)
-	go m.unstructured.Start(stop)
-	<-stop
+func (m *InformersMap) Start(ctx context.Context) error {
+	go m.structured.Start(ctx)
+	go m.unstructured.Start(ctx)
+	<-ctx.Done()
 	return nil
```

Comment:
Actually, just reverted this to still return nil, I'll open a different issue to tackle this in a different PR, but all the tests are expecting Start to return nil, and it doesn't take into account `context.Cancelled` error when the context is cancelled at the end of a test, does that sound good?

Reply:
I think that makes more sense, because usually the manager is started with something like
```
if err := mgr.Start(ctx); err != nil {
    log.Fatal(err, "manager failed")
}
```

even thought a canceled context is an expected condition.